File: amplify/backend/api/restapicontainer/src/app/lib/appsync.py
import requests
import json
import os
import boto3

api_key = os.environ.get('API_MARKETWATCH_GRAPHQLAPIKEYOUTPUT', None)

def query_api(query, variables: dict):
    endpoint = os.environ.get('API_MARKETWATCH_GRAPHQLAPIENDPOINTOUTPUT', None)
    headers={"Content-Type": "application/json"}
    headers["x-api-key"] = api_key
    payload = {
        "query": query, 
        'variables': variables
    }

    try:
        response = requests.post(
            endpoint,
            json=payload,
            headers=headers
        ).json()
        if 'errors' in response:
            print('Error attempting to query AppSync')
            print(response['errors'])
        else:
            return response
    except Exception as exception:
        print('Error with Mutation')
        print(exception)

    return None

File: amplify/backend/function/countDocuments/src/appsync.js
const https = require('https');
const urlParse = require("url").URL;
const AWS = require("aws-sdk");

async function executeQuery(query, variables) {
    const endpoint = new urlParse(process.env.API_MARKETWATCH_GRAPHQLAPIENDPOINTOUTPUT).hostname.toString();
    const req = new AWS.HttpRequest(process.env.API_MARKETWATCH_GRAPHQLAPIENDPOINTOUTPUT, process.env.REGION);
    req.method = "POST";
    req.path = "/graphql";
    req.headers.host = endpoint;
    req.headers["Content-Type"] = "application/json";
    let operation = {
		query: query,
		variables: variables
	};
    req.body = JSON.stringify(operation);
  
    req.headers["x-api-key"] = process.env.API_MARKETWATCH_GRAPHQLAPIKEYOUTPUT;

    const data = await new Promise((resolve, reject) => {
      const httpRequest = https.request({
        ...req,
        host: endpoint
      }, (result) => {
        let data = "";
  
        result.on("data", (chunk) => {
          data += chunk;
        });
  
        result.on("end", () => {
          resolve(JSON.parse(data.toString()));
        });
      });
  
      httpRequest.write(req.body);
      httpRequest.end();
    });
    return data;
};

module.exports.executeQuery = executeQuery;